### ISSUE TITLE
Cloud Server PAT authenticate error

### ISSUE THREAD

**Issue by AmiraLotfy** [sphinx-contrib/confluencebuilder#611]
I am using confluence server not cloud and based on the following link it was developed that we can create Personal Access Token which is similar to API token in confluence cloud to authenticate external application to be authenticated.
[https://jira.atlassian.com/browse/CONFSERVER-57389](url)

**sphinx-build-confluence report**
```
(system)
 platform: macOS-12.2-x86_64-i386-64bit
   python: 3.9.7 (default, Oct 13 2021, 06:45:31) [Clang 13.0.0 (clang-1300.0.29.3)]
   sphinx: 4.1.2
 requests: 2.25.1
  builder: 1.7.1

(configuration)
~default configuration~

(error loading configuration)
no documentation/missing configuration
```

I raised previously a question before regarding authentication with confluence in [issue 610](https://github.com/sphinx-contrib/confluencebuilder/issues/610)
 I can curl confluence rest api with my PAT but I can't do it with sphinx if there is anything missing in config that I didn't follow but I get authentication error on the same PAT token with sphinx

```

# Configuration file for the Sphinx documentation builder.
#
# This file only contains a selection of the most common options. For a full
# list see the documentation:
# https://www.sphinx-doc.org/en/master/usage/configuration.html

# -- Path setup --------------------------------------------------------------

# If extensions (or modules to document with autodoc) are in another directory,
# add these directories to sys.path here. If the directory is relative to the
# documentation root, use os.path.abspath to make it absolute, like shown here.
#
import os
import sys
sys.path.insert(0, os.path.abspath('../..'))
sys.path.insert(0, os.path.abspath('..'))


# -- Project information -----------------------------------------------------

project = 'data_css_webtracking'
copyright = '2021, Amira Lotfy'
author = 'Amira Lotfy'

# The full version, including alpha/beta/rc tags
release = '1.0'


# -- General configuration ---------------------------------------------------

# Add any Sphinx extension module names here, as strings. They can be
# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
# ones.
extensions = [
    'sphinx.ext.autodoc',
    'sphinx.ext.napoleon',
    'sphinx.ext.intersphinx',
    'sphinx.ext.ifconfig',
    'sphinx.ext.viewcode',
    'sphinx.ext.githubpages',
    'sphinxcontrib.confluencebuilder']
latex_elements = {
# The paper size ('letterpaper' or 'a4paper').
    'papersize': 'letterpaper',
# The font size ('10pt', '11pt' or '12pt').
    'pointsize': '10pt',
# Additional stuff for the LaTeX preamble.
    'preamble': '',
# Latex figure (float) alignment
    'figure_align': 'htbp',
}

# Add any paths that contain templates here, relative to this directory.
templates_path = ['_templates']

# List of patterns, relative to source directory, that match files and
# directories to ignore when looking for source files.
# This pattern also affects html_static_path and html_extra_path.
exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']


# -- Options for HTML output -------------------------------------------------

# The theme to use for HTML and HTML Help pages.  See the documentation for
# a list of builtin themes.
#
html_theme = 'classic'

# Add any paths that contain custom static files (such as style sheets) here,
# relative to this directory. They are copied after the builtin static files,
# so a file named "default.css" will overwrite the builtin "default.css".
html_static_path = ['_static']

confluence_publish = True
confluence_space_key = "XX"
confluence_parent_page = 'XXXXX'
confluence_page_hierarchy = True
# confluence_ask_password = True
# (for confluence server)
confluence_server_url = 'https://confluence.example.com/'
confluence_server_user = "USER"
confluence_server_pass = 'TOKEN'
```

**Comment by jdknight** [comment]
@AmiraLotfy, ah ~~ I honestly have only used/tested with the API tokens when they were first introduced on Cloud instances. There was an assumption that [personal access tokens (PATs)](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html) would be handled in a similar manner (just accepting a token value). Upon further inspection, it appears to not be the case. It appears to use PATs, the `Authorization` header must be set with the token value.

Fortunately, using PATs is possible by manually setting the header option using the [`confluence_publish_headers`](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/configuration/#confval-confluence_publish_headers) configuration. To authenticate with a PAT _do not set_ (or unset) `confluence_server_user`/`confluence_server_pass`, while using the following in a Sphinx configuration:

```
pat_token = 'MY_PAT_VALUE'
confluence_publish_headers = {
    'Authorization': 'Bearer ' + pat_token,
}
```

In the future, a `confluence_publish_pat` option will probably be introduced by this extension for convenience/clarity.

**Comment by AmiraLotfy** [comment]
Thank you so much, I can use this workaround until the option is added in further release.

**Comment by jdknight** [comment]
Changes have been added in #618 to improve users support for configuring personal access tokens (PATs). PATs (not to be confused with API tokens) will be able to configured by using the following:

```
confluence_publish_token = `my-token`
```

This change has been added into the main branch and should be made available next stable release (v1.8). Users wanting to use this option right away, the [development version](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/install/#development-installation) of this extension can be installed. Otherwise, the [workaround mentioned above](https://github.com/sphinx-contrib/confluencebuilder/issues/611#issuecomment-1030645274) can be used for the interim.

**Comment by jdknight** [comment]
v1.8 is now available on PyPI -- marking as closed.
